tests/end_to_end.rs

```diff
@@ -1071,3 +1072,172 @@ async fn test_vacuum_command() {
+    }
+
+    let context = make_context_with_pg().await;
+
+    // Creates table with table_versions 1 (empty) and 2
+    create_table_and_insert(&context, "test_table").await;
+
+    // Add another partition for table_version 3
+    let plan = context
+        .plan_query("INSERT INTO test_table (some_value) VALUES (45), (46), (47)")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    // Add another partition for table_version 4
+    let plan = context
+        .plan_query("INSERT INTO test_table (some_value) VALUES (46), (47), (48)")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    // Add another partition for table_version 5
+    let plan = context
+        .plan_query("INSERT INTO test_table (some_value) VALUES (42), (41), (40)")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    // We have 4 partitions from 4 INSERTS
+    assert_partition_ids(&context, 5, vec![1, 2, 3, 4]).await;
+
+    //
+    // Execute DELETE affecting partitions 2, 3 and creating table_version 6
+    //
+    let plan = context
+        .plan_query("DELETE FROM test_table WHERE some_value > 46")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 6, vec![1, 4, 5]).await;
+
+    let partitions = context
+        .partition_catalog
+        .load_table_partitions(6 as TableVersionId)
+        .await
+        .unwrap();
+
+    // Assert result of the new partition with id 5
+    let results = scan_partitions(&context, Some(vec![4]), partitions[2].clone()).await;
+    let expected = vec![
+        "+------------+",
+        "| some_value |",
+        "+------------+",
+        "| 45         |",
+        "| 46         |",
+        "| 46         |",
+        "+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    // Verify results
+    let plan = context
+        .plan_query("SELECT some_value FROM test_table")
+        .await
+        .unwrap();
+    let results = context.collect(plan).await.unwrap();
+
+    // TODO: the partition order is swapped here
+    let expected = vec![
+        "+------------+",
+        "| some_value |",
+        "+------------+",
+        "| 42         |",
+        "| 43         |",
+        "| 44         |",
+        "| 42         |",
+        "| 41         |",
+        "| 40         |",
+        "| 45         |",
+        "| 46         |",
+        "| 46         |",
+        "+------------+",
+    ];
+    assert_batches_eq!(expected, &results);
+
+    //
+    // Execute DELETE with multiple conditions, removing entire partition 4, and trimming partitions 1 and 5
+    //
+    let plan = context
+        .plan_query("DELETE FROM test_table WHERE some_value < 43 OR some_value > 45")
+        .await
+        .unwrap();
+    context.collect(plan).await.unwrap();
+
+    assert_partition_ids(&context, 7, vec![6]).await;
```

Comment:
Do we end up with one partition because the pruning doesn't discard any of `1, 4, 5` and so we end up collapsing them into a single partition? It would be good to have a test where we have a `(not-pruned partition, pruned partition, not-pruned partition, not-pruned partition)` to check that we end up with `(processed partition, pruned-so-stays-the-same partition, processed-partition-last-two-combined)`